### PR TITLE
Feature/texturewriter stb

### PR DESCRIPTION
**Pull request by WeirdRubberDuck**
Add an STB texture writer to ghoul and use that for generating the single color pixel image instead of a hardcoded PPM file (that did not work on AMD cards...)

Should close https://github.com/OpenSpace/OpenSpace/issues/1965 (but still have to verify on my AMD machine)

Note that this implementation currently only uses the PNG support of the writer, but that other file formats are also possible. I have tested it to make sure that it outputs the supported formats in the writer correctly.   

If you really feel like you want to test it yourself as well let me know and I can share the temporary Lua function I made to generate images with an arbitrary number of pixel values. 

Related PR in Ghoul: https://github.com/OpenSpace/Ghoul/pull/101

### CODE REVIEW

File: src/engine/openspaceengine_lua.inl [hunk 1]
```diff
@@ -136,40 +151,45 @@ namespace {
...
         unsigned int height = 1;
         unsigned int size = width * height;
-        std::vector<unsigned char> img(size * 3);
-        img[0] = static_cast<unsigned char>(255 * color.r);
-        img[1] = static_cast<unsigned char>(255 * color.g);
-        img[2] = static_cast<unsigned char>(255 * color.b);
+        std::vector<GLubyte> img(size * 3);
+        img[0] = static_cast<GLubyte>(255 * color.r);
+        img[1] = static_cast<GLubyte>(255 * color.g);
+        img[2] = static_cast<GLubyte>(255 * color.b);
```
Comment (alexanderbock) [inline src/engine/openspaceengine_lua.inl:169]:
```suggestion
        std::array<GLubyte, Size * 3> img = {
            static_cast<GLubyte>(255 * color.r),
            static_cast<GLubyte>(255 * color.g),
            static_cast<GLubyte>(255 * color.b)
        };
```

File: src/engine/openspaceengine_lua.inl [hunk 2]
```diff
@@ -136,40 +151,45 @@ namespace {
...
     else {
-        // Write the color to a ppm file
-        static std::mutex fileMutex;
-        std::lock_guard guard(fileMutex);
-        std::ofstream ppmFile(fileName, std::ofstream::binary | std::ofstream::trunc);
-
+        // Write the color to a new file
         unsigned int width = 1;
         unsigned int height = 1;
         unsigned int size = width * height;
```
Comment (alexanderbock) [inline src/engine/openspaceengine_lua.inl:165]:
```suggestion
        constexpr unsigned int Width = 1;
        constexpr unsigned int Height = 1;
        constexpr unsigned int Size = Width * Height;
```

File: src/engine/openspaceengine_lua.inl [hunk 3]
```diff
@@ -136,40 +151,45 @@ namespace {
...
-        img[2] = static_cast<unsigned char>(255 * color.b);
+        std::vector<GLubyte> img(size * 3);
+        img[0] = static_cast<GLubyte>(255 * color.r);
+        img[1] = static_cast<GLubyte>(255 * color.g);
+        img[2] = static_cast<GLubyte>(255 * color.b);
+
+        using Texture = ghoul::opengl::Texture;
+        Texture textureFromData = Texture(
+            reinterpret_cast<void*>(img.data()),
+            glm::uvec3(width, height, 1),
```
Comment (alexanderbock) [inline src/engine/openspaceengine_lua.inl:174]:
```suggestion
            glm::uvec3(Width, Height, 1),
```

File: src/engine/openspaceengine_lua.inl [hunk 4]
```diff
@@ -136,40 +151,45 @@ namespace {
...
+            GL_TEXTURE_2D,
+            Texture::Format::RGB
+        );
+        textureFromData.setDataOwnership(Texture::TakeOwnership::No);
 
-        if (!ppmFile.is_open()) {
-            throw ghoul::lua::LuaError("Could not open ppm file for writing");
+        try {
+            ghoul::io::TextureWriter::ref().saveTexture(textureFromData, fileName.string());
```
Comment (alexanderbock) [inline src/engine/openspaceengine_lua.inl:181]: Line length exceeded

**Review by alexanderbock — APPROVED** [review]

**Comment by alexanderbock** [comment]
No longer getting the linker errors

**Comment by GPayne** [comment]
Getting a build error on linux using gcc13:
```
[  2%] Building CXX object ext/ghoul/ext/assimp/code/CMakeFiles/assimp.dir/AssetLib/MDL/MDLLoader.cpp.o
In file included from /usr/include/c++/13/bits/hashtable_policy.h:36,
                 from /usr/include/c++/13/bits/hashtable.h:35,
                 from /usr/include/c++/13/bits/unordered_map.h:33,
                 from /usr/include/c++/13/unordered_map:41,
                 from /usr/include/c++/13/functional:63,
                 from /home/gene/Desktop/OpenSpace/build/ext/ghoul/ext/assimp/code/CMakeFiles/assimp.dir/cmake_pch.hxx:5,
                 from <command-line>:
In static member function ‘static _Up* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = long unsigned int; _Up = long unsigned int; bool _IsMove = false]’,
    inlined from ‘_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:506:30,
    inlined from ‘_OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:533:42,
    inlined from ‘_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:540:31,
    inlined from ‘_OI std::copy(_II, _II, _OI) [with _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:633:7,
    inlined from ‘std::vector<bool, _Alloc>::iterator std::vector<bool, _Alloc>::_M_copy_aligned(const_iterator, const_iterator, iterator) [with _Alloc = std::allocator<bool>]’ at /usr/include/c++/13/bits/stl_bvector.h:1305:28,
    inlined from ‘void std::vector<bool, _Alloc>::_M_reallocate(size_type) [with _Alloc = std::allocator<bool>]’ at /usr/include/c++/13/bits/vector.tcc:851:40,
    inlined from ‘void std::vector<bool, _Alloc>::reserve(size_type) [with _Alloc = std::allocator<bool>]’ at /usr/include/c++/13/bits/stl_bvector.h:1093:17,
    inlined from ‘Assimp::MDL::IntSharedData_MDL7::IntSharedData_MDL7()’ at /home/gene/Desktop/OpenSpace/ext/ghoul/ext/assimp/code/AssetLib/MDL/MDLFileData.h:879:32,
    inlined from ‘void Assimp::MDLImporter::InternReadFile_3DGS_MDL7()’ at /home/gene/Desktop/OpenSpace/ext/ghoul/ext/assimp/code/AssetLib/MDL/MDLLoader.cpp:1351:29:
/usr/include/c++/13/bits/stl_algobase.h:437:30: error: ‘void* __builtin_memmove(void*, const void*, long unsigned int)’ forming offset 8 is out of the bounds [0, 8] [-Werror=array-bounds=]
  437 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [ext/ghoul/ext/assimp/code/CMakeFiles/assimp.dir/build.make:1740: ext/ghoul/ext/assimp/code/CMakeFiles/assimp.dir/AssetLib/MDL/MDLLoader.cpp.o] Error 1
```

**Comment by GPayne** [comment]
Did a test with fresh clone and build directory on windows. Got this build error on line 64 of src/rendering/renderengine.cpp:
`Cannot open include file: 'ghoul/io/texture/texturewriterstb.h': No such file or directory`
I confirmed that I have commit 412e90e4 of submodule ghoul.

**Pull request merged**